addons/mail/models/mail_channel.py

```diff
@@ -322,6 +322,8 @@ def action_unfollow(self):
         return self._action_unfollow(self.env.user.partner_id)
 
     def _action_unfollow(self, partner):
+        if partner not in self.channel_partner_ids:
+            return False
```

Comment:
It returned the result of a 'write', which means keeping return True, even if it does not mean much.